src/dbnode/persist/fs/merger.go

```diff
@@ -229,23 +230,24 @@ func (m *merger) Merge(
 		// Closing the context will finalize the data returned from
 		// mergeWith.Read(), but is safe because it has already been persisted
 		// to disk.
-		tmpCtx.BlockingClose()
+		// NB(r): Make sure to use BlockingCloseReset so can reuse the context.
```

Comment:
isnt the reset handled by the reset right before the call to `mergeWith.Read()`?

Reply:
Yeah the main thing we avoid using `BlockingCloseReset()` is to avoid the context going back to the pool. (i.e. the Reset as the suffix means, after closing do not put back in the pool, we explicitly want to reuse this context).

And the main reason need to use a pooled context here is since the context needs to refer to the pooled finalizers linked list now, or else it will allocate a ton in between uses in this tight loop.